Feature/refactoring

ArkanEmre opened the pull request:
Refactoring and clean up of the code

Commit:
simplify load logic of RDFDatasetDirectStatementDeleter

Commit:
refactor code for better readability

Commit:
refactor code for better readability
add javadoc comments

Commit:
refactor and clean RDFEndpoint

Commit:
refactor and clean DirectStatementDeleter

Commit:
refactor and clean RDFDatasetDirectStatementLoader

Commit:
refactor and clean DirectStatementLoader

Commit:
refactor and clean RDFImport

Commit:
refactor and clean RDFParserConfig

Commit:
refactor and clean RDFToLPGStatementProcessor

Commit:
refactor and clean StatementPreviewer

Commit:
refactor and clean StatementStreamer

Commit:
refactor and clean RDFEndpointTest

Commit:
rename comparemodels to compareModels in ModelTestUtils

Commit:
refactor and clean RDFImportTest

Commit:
Merge branch '3.5' of https://github.com/jbarrasa/neosemantics into feature/refactoring

rnavarropiris commented:
All OK from my side :)
Since an [open pull request](https://github.com/jbarrasa/neosemantics/pull/96) from `develop` into the original repository is still open, I'll wait.

Otherwise the code in this request will be part of the mentioned pull request.

@ArkanEmre I think we did a mistake in the branching workflow. 
Probably we should have created pull requests always from the feature branches, so that if and once accepted, pulling the changes from the remote upstream into our own develop would effectively "merge" the changes.

This way, we can safely have multiple open pull requests.

We can adapt this new policy once the mentioned pull request is finished.

Commit:
fix compilation errors

Commit:
Merge branch 'develop' into feature/refactoring

# Conflicts:
#	src/main/java/semantics/extension/RDFEndpoint.java
#	src/test/java/semantics/RDFImportTest.java
#	src/test/java/semantics/extension/RDFEndpointTest.java

Commit:
simplify assertions
remove implicit array creation
replace anonymous functions with lambda

Commit:
rename methods and classes concerning named graphs
edit documentation accordingly
add preconditions to import, preview, delete methods in RDFImport

Commit:
Merge remote-tracking branch 'origin/develop' into feature/refactoring

# Conflicts:
#	src/main/java/semantics/RDFImport.java

Commit:
fix problems due to merge
